test/validation/api/timer/timer.c

```diff
@@ -107,18 +101,14 @@ typedef struct {
 } global_shared_mem_t;
 
 static global_shared_mem_t *global_mem;
+static odp_instance_t inst;
+static odp_timer_clk_src_t clk_src;
```

Comment:
I'd put also these into SHM allocated shared mem. Those could locate outside of global_shared_mem_t, which is memset to zero between test suite rounds.

```
typedef struct {
     odp_instance_t inst;
     odp_timer_clk_src_t clk_src;

    global_shared_mem_t global_mem;
    
} test_global_t;
```

Reply:
Why though?

Reply:
When reading the code, it's easier to maintain the context between local vs global variables. Also, "all globals through one struct"-pattern is used in other applications as well to tackle process vs thread mode difference in globals.

Reply:
This is now a little different from other tests anyway, and clk_src is special since it's our "suite loop" variable. Regardless, I put it in shm in v2. And added some 15 lines of `odp_timer_clk_src_t clk_src = test_global->clk_src;`, although I don't think it makes the code more readable.

I did not put inst in shm, since inst has a slightly longer lifetime than shm. Other validation tests also don't do that.